### ISSUE TITLE
Crash when editing time entries on personal page

### ISSUE THREAD

**Issue by ilario-gottardello** [dhanasingh/redmine_wktime#121]
If a user edit a time log on her personal page, clicking right mouse button and then edit, a crash occours. Redmine log says:

Started GET "/redmine/time_entries/6807/edit" for 172.16.8.200 at 2017-10-04 09:12:33 +0200
Processing by TimelogController#edit as HTML
  Parameters: {"id"=>"6807"}
  Current user: some.user (id=16)
Completed 500 Internal Server Error in 5ms (ActiveRecord: 1.3ms)

NoMethodError (undefined method `[]' for nil:NilClass):
  plugins/redmine_wktime/lib/timelogcontroller_patch.rb:254:in `find_time_entry'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'

The plugin version is 3.0.1
If the same user goes to the time entries page and tries to edit a time log from there, all works.

**Comment by dhanasingh** [comment]
can you try erpmine 3.1 and see if that fixes the problem

**Comment by ilario-gottardello** [comment]
I've upgraded to 3.1 but the problem still exists. The funny thing is that not all users are experiencing the bug

**Comment by dhanasingh** [comment]
ok, can you paste the error from the log ?

**Comment by ilario-gottardello** [comment]
Sure!

`Started GET "/redmine/time_entries/8802/edit" for 172.16.8.90 at 2017-12-19 17:57:30 +0100
Processing by TimelogController#edit as HTML
  Parameters: {"id"=>"8802"}
  Current user: some.user (id=6)
Completed 500 Internal Server Error in 6ms (ActiveRecord: 1.7ms)

NoMethodError (undefined method `[]' for nil:NilClass):
  plugins/redmine_wktime/lib/timelogcontroller_patch.rb:344:in 'find_time_entry'
  lib/redmine/sudo_mode.rb:63:in 'sudo_mode'`

**Comment by ilario-gottardello** [comment]
Environment:
  Redmine version                3.4.3.stable
  Ruby version                   2.3.1-p112 (2016-04-26) [x86_64-linux-gnu]
  Rails version                  4.2.8
  Environment                    production
  Database adapter               PostgreSQL
SCM:
  Subversion                     1.9.3
  Git                            2.7.4
  Filesystem                     
Redmine plugins:
  redmine_base_deface            0.0.1
  redmine_dmsf                   1.6.0
  redmine_issue_dynamic_edit     0.4.5
  redmine_refresh                0.2.0
  redmine_scm                    0.5.1
  redmine_wktime                 3.1

**Comment by ilario-gottardello** [comment]
Unfortunately, same error as before.. :(

```
Started GET "/redmine/time_entries/9852/edit" for 172.16.8.201 at 2018-02-07 13:45:59 +0100
Processing by TimelogController#edit as HTML
  Parameters: {"id"=>"9852"}
  Current user: some.user (id=26)
Completed 500 Internal Server Error in 5ms (ActiveRecord: 1.4ms)

NoMethodError (undefined method `[]' for nil:NilClass):
  plugins/redmine_wktime/lib/timelogcontroller_patch.rb:345:in `find_time_entry'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

**Comment by ilario-gottardello** [comment]
Environment:
  Redmine version                3.4.4.stable
  Ruby version                   2.3.1-p112 (2016-04-26) [x86_64-linux-gnu]
  Rails version                  4.2.8
  Environment                    production
  Database adapter               PostgreSQL
SCM:
  Subversion                     1.9.3
  Git                            2.7.4
  Filesystem                     
Redmine plugins:
  redmine_base_deface            0.0.1
  redmine_dmsf                   1.6.0
  redmine_issue_dynamic_edit     0.6.1
  redmine_refresh                0.2.0
  redmine_scm                    0.5.1
  redmine_wktime                 3.2

**Comment by dhanasingh** [comment]
ok, how do you directly go to the time entries edit page from the my page ?
We try to go to a spent time page from my page from spent time widget and Reported Issues widget, it still takes you to the list page and then from there you go to the detail page.
If you go to the list page and then to the edit page, you don't get this error

**Comment by ilario-gottardello** [comment]
Yes, it's the workaround we use now. The error appears if you, from the Spent Time widget on the Personal Page, click with the right mouse button on a time entry and choose edit. This is a shortcut to go to the edit time spent page. This passage works for some people but doesn't work for others, giving back the internal server error I reported to you

**Comment by ilario-gottardello** [comment]
How may I help you? How could I debug the issue?

**Comment by karthickmadhiazhagan** [comment]
No, I think we know what the problem is. One thing we need to found is why some people are not getting the error. Are they going to spent time list page at least once before going to the personal page?

**Comment by ilario-gottardello** [comment]
I usually leave open the personal page, and add time entries directly from there. If I need to edit something, I use right click, and it always works for me (I am an administrator, but other non-admin users do not experience the error). I never go to the time_entries page

**Comment by karthickmadhiazhagan** [comment]
No, It is not permission issue. It is related to the patch code of edit Time entries. If you are directly access the time entry edit page apart from spent time list page then patch code not working. We have fixed this issue. It will be available on our next release.

**Comment by ilario-gottardello** [comment]
Is it fixed on the 3.2.1 update or should I wait your next major release? It's a production server so upgrading too frequently it's not a good idea...

**Comment by dhanasingh** [comment]
no, will be available in the next release

**Comment by ilario-gottardello** [comment]
Hello, how is it going with the next release? Our users are eager to have this problem fixed... Thank you very much!

**Comment by ilario-gottardello** [comment]
Hello, sorry to bug you so much, but could you please give me some information about then release date? Or is it possible to issue a bugfix release like 3.2.2 or something? Thank you very much!

**Comment by ilario-gottardello** [comment]
Could you please instruct me how to backport it to 3.2.1? I really need to fix this...

**Comment by karthickmadhiazhagan** [comment]
Sorry for the delayed response. We have fixed this issue on 3.3. Hopefully it works fine. 
Can you please check?
